app.go

```diff
@@ -618,9 +667,16 @@ func (app *App) provide(p provide) {
 
 	switch {
 	case p.IsSupply:
-		app.logger.PrintSupply(constructor)
+		for _, rtype := range fxreflect.ReturnTypes(constructor) {
+			fxlog.Info("supplying", fxlog.Field{Key: "constructor", Value: rtype}).Write(app.log)
```

Comment:
the key should be type or returnType. `fx.Supply(v)` is used to provide direct values.

```suggestion
			fxlog.Info("supplying", fxlog.Field{Key: "type", Value: rtype}).Write(app.log)
```